### PR TITLE
vastdata driver: refactor driver; added python3 support; added trash api

### PR DESCRIPTION
**Pull request by vlboiko**

### CODE REVIEW

File: manila/share/drivers/vastdata/__init__.py [hunk 1]
```diff
@@ -334,41 +219,72 @@ def shrink_share(self, share, new_size, share_server=None):
...
+                yield mp
+            finally:
+                utils.execute("umount", mp, run_as_root=True)
+                utils.execute("rmdir", mp, run_as_root=True)
+
+        with _mounted_root() as mount:
+            src = self._to_volume_path(share_id, mount)
+            dst = f"{mount}/deleted/{share_id}"
+            utils.execute("mkdir", "-p", f"{mount}/deleted", run_as_root=True)
+            utils.execute("mv", src, dst, run_as_root=True)
```
Comment (koreno) [inline manila/share/drivers/vastdata/__init__.py]: let's drop support for local-mount deletion. we have no legacy to keep compatibility with here

File: manila/share/drivers/vastdata/__init__.py [hunk 2]
```diff
@@ -72,73 +71,60 @@
...
-            self.vms_session.post("views", data=data)
+        username = self.configuration.vast_mgmt_user
+        password = self.configuration.vast_mgmt_password
+        host = self.configuration.vast_mgmt_host
+        self.rest = RestApi(host, username, password, False, self.VERSION)
 
-        LOG.debug('setup complete')
+        if not self.rest.views.one(name=manila_view):
+            policy = self.rest.view_policies.one(name="default")
+            self.rest.views.create(name=manila_view, path=self._root_export, policy_id=policy.id)
```
Comment (koreno) [inline manila/share/drivers/vastdata/__init__.py]: any idea why we're doing this?...

**Review by koreno — CHANGES_REQUESTED** [review]

File: manila/share/drivers/vastdata/__init__.py (hunk 1 above)
Comment (vlboiko) [reply manila/share/drivers/vastdata/__init__.py]: Done

File: manila/share/drivers/vastdata/__init__.py (hunk 2 above)
Comment (vlboiko) [reply manila/share/drivers/vastdata/__init__.py]: For local mounting of parent directory before deletion.
I deleted local mount in favor of using Trash API so this part is also deleted.

**Pull request closed without merging**